test/extractor/ConfigurationExtractor.test.ts

```diff
@@ -0,0 +1,128 @@
+    const codeSystemUrl = cloneDeep(profileUrl);
+    codeSystemUrl.value = 'http://example.org/our/ig/CodeSystem/TypicalCodes';
+    const codeSystemStatus = cloneDeep(extensionStatus);
+    codeSystemStatus.value = new fshtypes.FshCode('active');
+    typicalCodeSystem.rules.push(
+      codeSystemUrl,
+      codeSystemStatus,
+      cloneDeep(extensionVersion),
+      cloneDeep(extensionFhirVersion)
+    );
+
+    resources.add(typicalProfile);
+    resources.add(typicalExtension);
+    resources.add(typicalValueSet);
+    resources.add(typicalCodeSystem);
+  });
+
+  describe('#process', () => {
+    it('should return a configuration with a canonical and fhirVersion when there are no resources', () => {
+      const emptyResources = new Package();
+      const result = ConfigurationExtractor.process(emptyResources);
+      expect(result.config.canonical).toBeDefined();
+      expect(result.config.fhirVersion).toBeDefined();
+    });
+
+    it('should return a configuration with elements inferred from the canonical', () => {
+      const result = ConfigurationExtractor.process(resources);
+      expect(result.config.canonical).toBe('http://example.org/our/ig');
+      expect(result.config.id).toBe('example.our.ig');
+      expect(result.config.name).toBe('ExampleOurIg');
+    });
+
+    it('should return a configuration with elements inferred from resources', () => {
+      const result = ConfigurationExtractor.process(resources);
+      expect(result.config.fhirVersion).toEqual(['4.0.1']);
+      expect(result.config.version).toBe('0.8.6');
+      expect(result.config.status).toBe('draft');
+    });
+  });
+
+  describe('#inferCanonical', () => {
+    it('should return an empty string when the input contains no resources', () => {
+      const emptyResources = new Package();
+      const result = ConfigurationExtractor.inferCanonical(emptyResources);
+      expect(result).toBe('');
+    });
+
+    it('should return an empty string when none of the resources have a defined url', () => {
+      const noUrlResources = cloneDeep(resources);
+      // replace the rules that set a url with some empty rules
+      const differentRule = new ExportableCaretValueRule('');
+      differentRule.caretPath = 'publisher';
+      differentRule.value = 'Somebody';
+      noUrlResources.profiles[0].rules[0] = differentRule;
+      noUrlResources.extensions[0].rules[0] = cloneDeep(differentRule);
+      noUrlResources.valueSets[0].rules[0] = cloneDeep(differentRule);
+      noUrlResources.codeSystems[0].rules[0] = cloneDeep(differentRule);
+      const result = ConfigurationExtractor.inferCanonical(noUrlResources);
+      expect(result).toBe('');
+    });
+
+    it('should use the common url base to determine the canonical when all resources have the same url base', () => {
+      const result = ConfigurationExtractor.inferCanonical(resources);
+      expect(result).toBe('http://example.org/our/ig');
+    });
```

Comment:
I think it might be nice to have another test or two that demonstrates what happens w/ URLs w/ more or less path parts, for example, `http://example.org` and `http://example.org/this/is/our/ig`.  Hard to say for sure, but I think I'd expect `example` and `example.this.is.our.ig` respectively.